projects/packages/publicize/src/class-publicize.php

```diff
@@ -769,20 +769,12 @@ public function get_post_share_status( $post_id ) {
 
 		if ( $done ) {
 			// The site could have multiple admins, editors and authors connected. Load shares information that only the current user has access to.
-			$connection_ids = array_map(
-				function ( $connection ) {
-					if ( isset( $connection['connection_id'] ) ) {
-						return (int) $connection['connection_id'];
-					}
-					return 0;
-				},
-				$this->get_all_connections_for_user()
-			);
+			$connection_ids = wp_list_pluck( Connections::get_all_for_user(), 'connection_id', 'connection_id' );
```

Comment:
Is the casting to int not needed anymore?

Reply:
No, it's not needed because now the check is for array key instead of value and array keys work both as strings and numbers.

Reply:
> The key can either be an int or a string.
> Strings containing valid decimal ints, unless the number is preceded by a + sign, will be cast to the int type.

[PHP Docs](https://www.php.net/manual/en/language.types.array.php#language.types.array.syntax)